action/src/app.ts

```diff
@@ -60,52 +61,43 @@ async function run(stoatConfig: any) {
   const ghBranch = core.getInput('pr_branch_name');
   await waitForStoatDevServer(github.context.repo, ghBranch, repoSha);
 
-  core.info('Checking if prior steps succeeded...');
-  let stepsSucceeded = true;
+  core.info(`Fetching commit timestamp...`);
+  const ghCommitTimestamp = await getGhCommitTimestamp(octokit, github.context.repo, repoSha);
 
+  // find the current job
   const jobListResponse = await octokit.rest.actions.listJobsForWorkflowRun({
     owner: github.context.repo.owner,
     repo: github.context.repo.repo,
     run_id: github.context.runId
   });
-
-  // todo: in the future we may want to determine which job we're currently in
-  // with matrix jobs and such this can be difficult to determine
-  // see https://github.com/actions/toolkit/issues/550 and the other plethora of issues complaining about this
-  for (const job of jobListResponse.data.jobs) {
-    core.info(`Inspecting job "${job.name}"`);
-    for (const step of job.steps || []) {
-      core.info(`-- Step "${step.name}": ${step.conclusion}`);
-      if (step.conclusion !== null && step.conclusion !== 'skipped') {
-        stepsSucceeded = stepsSucceeded && step.conclusion === 'success';
-      }
-    }
-  }
-
-  core.info(`Prior steps succeeded: ${stepsSucceeded}`);
-
-  core.info(`Fetching commit timestamp...`);
-  const ghCommitTimestamp = await getGhCommitTimestamp(octokit, github.context.repo, repoSha);
-  // The context.job in @actions/github is GITHUB_JOB, which is the job id, not the name.
-  // It is different from the job name in the job list response. So we cannot use it to
-  // search for the job information. We use job run id instead.
-  // References:
-  // https://github.com/actions/toolkit/blob/main/packages/github/src/context.ts
-  // https://docs.github.com/en/actions/learn-github-actions/environment-variables
   const ghJobId = github.context.job;
   const ghJobRunId = github.context.runId;
-  const ghJob: GithubJob | undefined = jobListResponse.data.jobs.find((j) => j.run_id === ghJobRunId);
-
-  if (ghJob === undefined) {
+  // There is no precise way to find the current running job. To do that,
+  // we need to use an identifier from the github.context and search for
+  // it in the job list. Usually this works, because both github.context.job
+  // and job.name are job ids. However, when the job has a custom name or
+  // there are matrix variants, job.name refers to the custom name or a name
+  // with the matrix variants. In those cases, nothing from github.context
+  // can be used to find the job.
+  const ghJob: GithubJob | undefined = jobListResponse.data.jobs.find(
+    (j) => j.run_id === ghJobRunId && j.status === 'in_progress'
```

Comment:
@jrhizor, I think this is always true when there is no matrix involved.

Reply:
Was there a reason we couldn't pull the value from the job context object? something like https://github.com/stoat-dev/stoat-action/actions/runs/3910711157/jobs/6683230548 seems to pull the correct value for the specific job without attempting to match ids; I imagine it would work for matrix builds as well? 

I think the reason we didn't do that before was because we chose to decide status at the workflow-level which required all jobs and steps to decide (because we couldn't access any of their actual job context objects)?

Reply:
To determine job status, this is a great approach. However, we still need the `ghJob` object for the job runtime plugin currently. So locating the running job is still required. (In the future, we may be able to move away from this.)

Reply:
Will use your approach.